src/debug-generator/debug-generator.c

```diff
@@ -158,8 +163,77 @@ static void install_debug_shell_dropin(void) {
                 log_warning_errno(r, "Failed to write drop-in for debug-shell.service, ignoring: %m");
 }
 
+static int process_unit_credentials(const char *credentials_dir) {
+        _cleanup_close_ int fd = -EBADF;
+        int r;
+
+        assert(credentials_dir);
+
+        fd = open(credentials_dir, O_CLOEXEC|O_DIRECTORY);
+        if (IN_SET(errno, ENXIO, ENOENT)) /* Credential env var not set, or dir doesn't exist. */
```

Comment:
Hmm? This must be in the `fd < 0` branch, otherwise we might be seeing `errno` from previous functions?